packages/new-editor/src/components/editor/variables/detail/Variable.tsx

```diff
@@ -40,9 +39,9 @@ export const VariableDetail = ({ variables, variablePath, setVariables }: Variab
       {!hasChildren && (
         <Fieldset label='Metadata'>
           <SimpleSelect
-            value={selectedMetadataOption?.value}
+            value={variable.metadata}
             items={metadataOptions}
-            onValueChange={value => handleVariableAttributeChange('metadata', value)}
+            onValueChange={(value: (typeof metadataOptions)[number]['value']) => handleVariableAttributeChange('metadata', value)}
```

Comment:
I recommend you to create a type in the Variables.ts to do this:
```ts
type VariableMetadata: (typeof metadataOptions)[number]['value'];
```
and use it simple here and in the Variable itself